### PR TITLE
Add TELEPORT_TYPE EventContext for Enderman and /teleport teleportations

### PR DESCRIPTION
**Pull request by RedNesto**
The only remaining TELEPORT_TYPE I see is #1629 after #1741 and this PR.

### CODE REVIEW

**Comment by phit** [comment]
~qa

File: src/main/java/org/spongepowered/common/mixin/core/command/server/MixinCommandTeleport.java [hunk 1]
```diff
@@ -133,7 +136,9 @@ private static void doTeleport(Entity p_189862_0_, CommandBase.CoordinateArg p_1
             double x = p_189862_1_.getAmount();
             double y = p_189862_2_.getAmount();
             double z = p_189862_3_.getAmount();
+            Sponge.getCauseStackManager().addContext(EventContextKeys.TELEPORT_TYPE, TeleportTypes.COMMAND);
```
Comment (Zidane) [inline src/main/java/org/spongepowered/common/mixin/core/command/server/MixinCommandTeleport.java]: You should wrap this in a cause stack frame and don't remove from the context (as the frame would be closed)

File: src/main/java/org/spongepowered/common/mixin/core/entity/monster/MixinEntityEnderman.java [hunk 2]
```diff
@@ -25,16 +25,30 @@
...
+import org.spongepowered.asm.mixin.injection.Redirect;
 
 import java.util.List;
 
 @Mixin(EntityEnderman.class)
 public abstract class MixinEntityEnderman extends MixinEntityMob implements Enderman {
 
+    @Redirect(method = "teleportTo", at = @At(value = "INVOKE", target = "Lnet/minecraft/entity/monster/EntityEnderman;attemptTeleport(DDD)Z"))
+    public boolean redirectTeleportTo(EntityEnderman entityEnderman, double x, double y, double z) {
+        Sponge.getCauseStackManager().addContext(EventContextKeys.TELEPORT_TYPE, TeleportTypes.ENTITY_TELEPORT);
```
Comment (Zidane) [inline src/main/java/org/spongepowered/common/mixin/core/entity/monster/MixinEntityEnderman.java]: Same here, wrap in a CauseStackFrame.

**Comment by gabizou** [comment]
Fix your commits, no need for the merge with bleeding commit.

**Comment by RedNesto** [comment]
@gabizou I cleaned the commits, first time I did it, hope I didn't missed something.

**Review by gabizou — APPROVED** [review]

**Pull request merged**